test/integration/scheduler_perf/scheduler_test.go

```diff
@@ -253,3 +238,107 @@ func schedulePods(config *testConfig) int32 {
+func (na nodeAffinity) mutatePodSpec(numPods int, pod *v1.Pod) []*v1.Pod {
+	numGroups := na.numGroups
+	nodeAffinityKey := na.nodeAffinityKey
+	podList := make([]*v1.Pod, 0, numGroups)
+	for i := 0; i < numGroups; i++ {
+		pod = &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "sched-perf-node-affinity-pod-",
+			},
+			Spec: testutils.MakePodSpec(),
+		}
+		pod.Spec.Affinity = &v1.Affinity{
+			NodeAffinity: &v1.NodeAffinity{
+				RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+					NodeSelectorTerms: []v1.NodeSelectorTerm{
+						{
+							MatchExpressions: []v1.NodeSelectorRequirement{
+								{
+									Key:      nodeAffinityKey,
+									Operator: v1.NodeSelectorOpIn,
+									Values:   []string{fmt.Sprintf("%v", i)},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+		podList = append(podList, pod)
+	}
+	return podList
+}
+
+// generatePodAndNodeTopology is the wrapper function for modifying both pods and node objects.
+func (inputConfig *schedulerPerfConfig) generatePodAndNodeTopology(config *testConfig) {
+	nodeAffinity := inputConfig.NodeAffinity
+	podCreatorConfig := testutils.NewTestPodCreatorConfig()
+	var nodeStrategies []testutils.CountToStrategy
+	var pod *v1.Pod
+	var podList []*v1.Pod
+	if nodeAffinity != nil {
+		// Mutate Node
+		nodeStrategies = nodeAffinity.mutateNodeSpec(config.numNodes)
+		// Mutate Pod TODO: Make this to return to podSpec.
+		podList = nodeAffinity.mutatePodSpec(config.numPods, pod)
+		numGroups := nodeAffinity.numGroups
+		for _, pod := range podList {
+			podCreatorConfig.AddStrategy("sched-perf-node-affinity", config.numPods/numGroups,
+				testutils.NewCustomCreatePodStrategy(pod),
+			)
+		}
+		config.nodePreparer = framework.NewIntegrationTestNodePreparer(
+			config.schedulerSupportFunctions.GetClient(),
+			nodeStrategies, "scheduler-perf-")
+		config.podCreator = testutils.NewTestPodCreator(config.schedulerSupportFunctions.GetClient(), podCreatorConfig)
+		// TODO: other predicates/priorities will be processed in subsequent if statements.
+	} else {
+		// Default configuration.
+		nodePreparer := framework.NewIntegrationTestNodePreparer(
+			config.schedulerSupportFunctions.GetClient(),
+			[]testutils.CountToStrategy{{Count: config.numNodes, Strategy: &testutils.TrivialNodePrepareStrategy{}}},
+			"scheduler-perf-",
+		)
+
+		podConfig := testutils.NewTestPodCreatorConfig()
+		podConfig.AddStrategy("sched-test", config.numPods, testutils.NewSimpleWithControllerCreatePodStrategy("rc1"))
+		podCreator := testutils.NewTestPodCreator(config.schedulerSupportFunctions.GetClient(), podConfig)
+		config.nodePreparer = nodePreparer
+		config.podCreator = podCreator
+	}
+	return
+}
+
+// parseInput reads a configuration and then applies it to a test configuration.
+func parseInput(config *testConfig) {
+	// High Level structure that should be filled for every predicate or priority.
+	inputConfig := &schedulerPerfConfig{
+		NodeAffinity: &nodeAffinity{
```

Comment:
- I still feel like this function isn't parsing anything, so we should delete it.
- If we don't delete it though lets at least remove the `sched-perf-node-affinity` key defaults, since those should be in read in from the test config.